client/my-sites/comments/comment/index.jsx

```diff
@@ -37,60 +37,49 @@ export class Comment extends Component {
 
 	state = {
 		isEditMode: false,
-		isExpanded: false,
 		isReplyVisible: false,
 	};
 
 	componentWillReceiveProps( nextProps ) {
-		if ( nextProps.isBulkMode && ! this.props.isBulkMode ) {
-			this.setState( { isExpanded: false } );
-		}
+		const { isBulkMode: wasBulkMode } = this.props;
+		const { isBulkMode } = nextProps;
+		const { isEditMode: wasEditMode, isReplyVisible: wasReplyVisible } = this.state;
+
+		this.setState( {
+			isEditMode: wasBulkMode !== isBulkMode ? false : wasEditMode,
+			isReplyVisible: wasBulkMode !== isBulkMode ? false : wasReplyVisible,
+		} );
 	}
 
 	storeCardRef = card => ( this.commentCard = card );
 
 	keyDownHandler = event => {
 		const { isBulkMode } = this.props;
-		const { isEditMode, isExpanded } = this.state;
-
 		const commentHasFocus =
 			document &&
 			this.commentCard &&
 			document.activeElement === ReactDom.findDOMNode( this.commentCard );
 
-		if ( isEditMode || ( isExpanded && ! commentHasFocus ) ) {
+		if ( ! isBulkMode || ! commentHasFocus ) {
 			return;
 		}
 
 		switch ( event.keyCode ) {
 			case 13: // enter
 			case 32: // space
 				event.preventDefault();
-				return isBulkMode ? this.toggleSelected() : this.toggleExpanded();
+				return this.toggleSelected();
 		}
 	};
 
 	toggleEditMode = () =>
 		this.setState( ( { isEditMode } ) => ( {
 			isEditMode: ! isEditMode,
-			isExpanded: ! isEditMode,
 			isReplyVisible: false,
 		} ) );
 
-	toggleExpanded = () => {
-		if ( ! this.props.isLoading && ! this.state.isEditMode ) {
-			this.setState( ( { isExpanded } ) => ( {
-				isExpanded: ! isExpanded,
-				isReplyVisible: false,
-			} ) );
-		}
-	};
-
 	toggleReply = () =>
-		this.setState( ( { isReplyVisible } ) => ( {
-			isExpanded: true,
-			isReplyVisible: ! isReplyVisible,
-		} ) );
+		this.setState( ( { isReplyVisible } ) => ( { isReplyVisible: ! isReplyVisible } ) );
```

Comment:
Nitpick: we can simplify this further by passing an object instead of a function:

`this.setState( { isReplyVisible: ! isReplyVisible } );`

Reply:
Nope!
Here the reason: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous 🙂

Reply:
Oh right :+1: